src/templates/base-component.ts

```diff
@@ -0,0 +1,72 @@
+export class BaseComponent<T extends keyof HTMLElementTagNameMap = keyof HTMLElementTagNameMap> {
+  protected node: HTMLElementTagNameMap[T];
+  protected children: BaseComponent[] = [];
+
+  constructor(
+    tagName: T,
+    props: Partial<HTMLElementTagNameMap[typeof tagName]> = {},
+    parentNode?: HTMLElement,
+  ) {
+    this.node = document.createElement(tagName);
+
+    if (props) {
+      Object.assign(this.node, props);
+    }
+
+    if (parentNode) {
+      parentNode.append(this.node);
+    }
+  }
+
+  getNode<K extends HTMLElementTagNameMap[T]>(): K {
+    return this.node as K;
+  }
+
+  getChildren(): BaseComponent[] {
+    return this.children;
+  }
+
+  addClass(...className: string[]) {
+    this.node.classList.add(...className);
+  }
+
+  removeClass(...className: string[]) {
+    this.node.classList.remove(...className);
+  }
+
+  toggleClass(className: string) {
+    this.node.classList.toggle(className);
+  }
+
+  append(...rest: BaseComponent[]): void {
+    rest.forEach((child) => {
+      this.children.push(child);
+      this.node.append(child.getNode());
+    });
+  }
+
+  setContent(content: string): void {
+    this.node.textContent = content;
+  }
+
+  setInnerHTML(html: string): void {
+    this.node.innerHTML = html;
```

Comment:
innerHTML is rather [dangerous](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#security_considerations), check the alternatives, please.

Reply:
- Element.setHTML() - don't have compatible with Browsers
- Other ways are remove and append elements. But I don't understand how I can insert static HTML instead of creating components.